### ISSUE TITLE
Farming Station don't farm anymore

### ISSUE THREAD

**Issue by MelanX** [HenryLoenwind/EnderIOAddons#45]
Minecraft Version: 1.7.10
Forge: 10.13.4.1614
Mod: 0.10.11
(EnderIO: 2.3.0.429)

Modules in the machine: 
- Seed Harvesting Module
- Plant Upgrading Module
- Seed Sorting Module
- Breeding Module
- Harvest Unanalyzed Plants Module
- Seed Analyzer Module

Sometimes the Farming Station work only at one seed. For many minutes and don't harvest anything. If I destroy the crop sticks, the machine plants new ones and does the job normal. Few minutes later there is the same problem. 
The problem is with watered and unwatered farmland. I use only crop sticks, diamond hoe and the modules. I have enough energy.
Don't know if you need more information or how I can describe better.

**Comment by MelanX** [comment]
I removed the Plant Upgrading Module now and the machine is now working well for five minutes. Since I removed it, no problems.

**Comment by HenryLoenwind** [comment]
more data: https://www.reddit.com/r/feedthebeast/comments/5x44ni/project_ozone_2_agricraft_farming_station_gets/

**Comment by HenryLoenwind** [comment]
Just one thought: Was it hanging on trying to do "plant upgrading" on a fully grown plant?

**Comment by MelanX** [comment]
Yes, it was. I reduced the delay for the machine in config and put the upgrade back. It was hanging but after few seconds, it was doing the job.
So, I have no big problem anymore but think, it's a bug.

**Comment by damaskinos** [comment]
I have the same problem that @MelanX  had.

What delay you changed? Cuz I tried and nothing, keeps "working" on the same grow plant and stuck

**Comment by HenryLoenwind** [comment]
Could you please enable the config setting `farmDebugLoggingEnabled` and post the relevant lines from the logfile? Please narrow it down to the lines talking about the plant where it is hanging (`"WorkTile [bc="` xyz-coordinates),

**Comment by rashek4** [comment]
I'm also having this issue. For me it's only the "Plant Upgrading Module" that's not working. The "Harvest Unanalyzed Plants Module" works fine. It only gets stuck when trying to remove plants that are analyzed. Without the "Plant Upgrading Module" everything seems to work as intended. I'm also on Project Ozone.
This is they logfile, I tried to trim it down to the relevant parts. 

https://www.dropbox.com/s/btedvkd1gs6kxj8/LogFarmingStation.txt?dl=0

**Comment by HenryLoenwind** [comment]
Thanks, I see what's happening:

The ReplaceBetterModule (a.k.a. plant upgrading module) says to destroy the plant, place fresh crops and plant a new seed. Then the HarvestUnanalyzedModule says that the plant should not be destroyed. Next the placing of crops and the planting fails because the old plant has not be destroyed.

I'll upload a new version shortly where the HarvestUnanalyzedModule does not reset the "doDestroy" value but adds to it if it sees an unanalyzed plant.

Update: Version 0.10.13 is uploaded and should be visible on curseforge in a couple of minutes.

**Comment by rashek4** [comment]
Thanks a lot for the quick fix, it works great for me now.
